phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -1987,10 +1993,37 @@ private PTable createTableInternal(CreateTableStatement statement, byte[][] spli
             int baseTableColumnCount =
                     tableType == PTableType.VIEW ? parent.getColumns().size()
                             : QueryConstants.BASE_TABLE_BASE_COLUMN_COUNT;
+
+            Long viewTTL = VIEW_TTL_NOT_DEFINED;
+            Long viewTTLHighWaterMark = MIN_VIEW_TTL_HWM;
+            Long viewTTLProp = (Long) TableProperty.VIEW_TTL.getValue(tableProps);;
+
+            if (tableType == VIEW && viewTTLProp != null && parentPhysicalName != null) {
```

Comment:
@jpisaac Shouldn't we also do some checks on the actual value of the `viewTTLProp` provided by the user, such as checking that it is non-negative, other bounds, etc.?